tests/truth_tests.bzl

```diff
@@ -509,6 +509,35 @@ def _collection_contains_exactly_test(env, _target):
         env = env,
         msg = "check multiplicity; expected with multiple, expected with unique",
     )
+
+    subject = truth.expect(fake_env).that_collection(["a", "d", "c", "b", "e"])
+    order = subject.contains_exactly(["a", "b", "c", "d", "e"])
+    _assert_no_failures(
+        fake_env,
+        env = env,
+        msg = "check same elements with expected in different order",
+    )
+    order.in_order()
+    _assert_failure(
+        fake_env,
+        [
+            "expected values all found, but with incorrect order:",
+            "0: a found at offset 0",
+            "1: b found at offset 3",
+            "2: c found at offset 2",
+            "3: d found at offset 1",
+            "4: e found at offset 4",
+            "actual values:",
+            "0: a",
+            "1: b",
+            "2: c",
+            "3: d",
+            "4: e",
```

Comment:
> Just curious -- what was causing the wrong order of actual values being shown? I was couldn't figure it out from the diff.

I think that `1: b` just ended up matching `1: b found at offset 3` since the assert applies to the entire stringified failure message.